components/director/internal/domain/tenant/resolver_test.go

```diff
@@ -0,0 +1,124 @@
+		TxFn           func() (*persistenceautomock.PersistenceTx, *persistenceautomock.Transactioner)
+		TenantSvcFn    func() *automock.BusinessTenantMappingService
+		TenantConvFn   func() *automock.BusinessTenantMappingConverter
+		ExpectedOutput []*graphql.Tenant
+		ExpectedError  error
+	}{
+		{
+			Name: "Success",
+			TxFn: txGen.ThatSucceeds,
+			TenantSvcFn: func() *automock.BusinessTenantMappingService {
+				TenantSvc := &automock.BusinessTenantMappingService{}
+				TenantSvc.On("List", txtest.CtxWithDBMatcher()).Return(modelTenants, nil).Once()
+				return TenantSvc
+			},
+			TenantConvFn: func() *automock.BusinessTenantMappingConverter {
+				TenantConv := &automock.BusinessTenantMappingConverter{}
+				TenantConv.On("MultipleToGraphQL", modelTenants).Return(graphqlTenants).Once()
+				return TenantConv
+			},
+			ExpectedOutput: graphqlTenants,
+		},
+		{
+			Name: "Returns error when getting tenants failed",
+			TxFn: txGen.ThatDoesntExpectCommit,
+			TenantSvcFn: func() *automock.BusinessTenantMappingService {
+				TenantSvc := &automock.BusinessTenantMappingService{}
+				TenantSvc.On("List", txtest.CtxWithDBMatcher()).Return(nil, testError).Once()
+				return TenantSvc
+			},
+			TenantConvFn: func() *automock.BusinessTenantMappingConverter {
+				TenantConv := &automock.BusinessTenantMappingConverter{}
+				return TenantConv
+			},
+			ExpectedError: testError,
+		},
+		{
+			Name: "Returns error when failing on begin",
+			TxFn: txGen.ThatFailsOnBegin,
+			TenantSvcFn: func() *automock.BusinessTenantMappingService {
+				TenantSvc := &automock.BusinessTenantMappingService{}
+				return TenantSvc
+			},
+			TenantConvFn: func() *automock.BusinessTenantMappingConverter {
+				TenantConv := &automock.BusinessTenantMappingConverter{}
+				return TenantConv
+			},
+			ExpectedError: testError,
+		},
+		{
+			Name: "Returns error when failing on commit",
+			TxFn: txGen.ThatFailsOnCommit,
+			TenantSvcFn: func() *automock.BusinessTenantMappingService {
+				TenantSvc := &automock.BusinessTenantMappingService{}
+				TenantSvc.On("List", txtest.CtxWithDBMatcher()).Return(modelTenants, nil).Once()
+				return TenantSvc
+			},
+			TenantConvFn: func() *automock.BusinessTenantMappingConverter {
+				TenantConv := &automock.BusinessTenantMappingConverter{}
+				return TenantConv
+			},
+			ExpectedError: testError,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			tenantSvc := testCase.TenantSvcFn()
+			tenantConv := testCase.TenantConvFn()
+			persist, transact := testCase.TxFn()
+			resolver := tenant.NewResolver(transact, tenantSvc, tenantConv)
+
+			// WHEN
+			result, err := resolver.Tenants(ctx)
+
+			// THEN
+			if testCase.ExpectedError != nil {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), testCase.ExpectedError.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, testCase.ExpectedOutput, result)
+
+			persist.AssertExpectations(t)
```

Comment:
This can be replaced with
```suggestion
			mock.AssertExpectationsForObjects(t, persist, transact, tenantSvc, tenantConv)
```